packages/fxa-auth-server/lib/subhub/updates.js

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+
+'use strict';
+
+const Joi = require('joi');
+
+const MESSAGE_SCHEMA = Joi.object().keys({
+  uid: Joi.string().required(),
+  active: Joi.boolean().required(),
+  subscriptionId: Joi.string().required(),
+  productName: Joi.string(),
+  eventId: Joi.string(),
+  eventCreatedAt: Joi.number().integer(),
+  messageCreatedAt: Joi.number().integer().required(),
+  del: Joi.any(), // a method (function) that's added to messages
+});
+
+function validateMessage(message) {
+  return Joi.validate(message, MESSAGE_SCHEMA);
+}
+
+module.exports = function (log) {
+
+  return function start(messageQueue, db) {
+
+    async function handleSubHubUpdates(message) {
+      const uid = message && message.uid;
+
+      log.info('handleSubHubUpdated', { uid, action: 'notify' });
+
+      const result = validateMessage(message);
+      if (result.error) {
+        log.error('handleSubHubUpdates', { uid, action: 'validate', validationError: result.error });
+        message.del();
+        return;
+      }
+
+      try {
+        if (message.active) {
+          await db.createAccountSubscription(uid, message.subscriptionId, message.productName, message.eventCreatedAt);
```

Comment:
I am mostly guessing that `eventCreatedAt` is the right value for `createdAt`, but `messageCreatedAt` is another candidate

Reply:
We should use the Stripe timestamp as the authoritative, which is `eventCreatedAt`, so that's correct here.